<a href="https://colab.research.google.com/github/dgavieira/NLP-Celero/blob/exploratory-analysis/NLP_Teste.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Upload do Dataset

In [1]:
import pandas as pd

## Upload de dados de teste


### Teste - Sentimento NEGATIVO

In [19]:
df_test_neg = pd.read_csv('https://drive.google.com/uc?id=1tUxBvBO_6wOEGe3fgxmU2wCDuUnqNeot', encoding='utf8', header=None)
df_test_neg = df_test_neg.T # transpondo o dataframe para ficar no formato usual
df_test_neg

0
0   Once again Mr. Costner has dragged out a movie...
1   Years ago, when DARLING LILI played on TV, it ...
2   Julie Andrews satirically prods her own goody-...
3   Blake Edwards tried very hard to change Julie ...
4   An obvious vanity press for Julie in her first...
5   Some may go for a film like this but I most as...
6   Few movies have dashed expectations and upset ...
7   For years, I've been a big fan of Park's work ...
8   Might end up being the biggest disappointment ...
9   Bloody Birthday opens to a shot of Meadowvale ...
10  This movie cannot be serious because it has a ...
11  This is a pale imitation of 'Officer and a Gen...
12  A mean spirited, repulsive horror film about 3...
13  Bloody Birthday is a totally rubbish slasher m...
14  When you make a film with a killer-kids premis...
15  When people say children are annoying u think ...
16  'Bloody Birthday' is an odd and, at times, hum...
17  Is it a poorly acted, cliche-ridden pile of tr...
18  Formulaic slasher film, only this one stars th...
19  Cover Girl is a lacklustre WWII musical with a...
20  Rita Hayworth plays a Brooklyn nightclub dance...
21  Like 'Singin' in the Rain', 'Cover Girl' has a...
22  It seems ever since 1982, about every two or t...
23  Rita Hayworth is just stunning at times and, f...
24  It hurt to watch this movie, it really did... ...
25  This film seems well made, and more efforts sh...
26  Sorry, but Jacqueline Hyde (get it??? - Jack L...
27  Jacqueline Hyde starts like any other normal d...
28  Maniratnam, who in India, is often compared wi...
29  This film, which I rented under the title "Bla...
30  The first 20 minutes were a little fun because...
31  I only came here to check Terror Hospital for ...
32  This is a hard film to rate. While it truly de...
33  Wow, another Kevin Costner hero movie. Postman...
34  I bought this on VHS as "Terror Hospital", and...
35  The film opens with a cult leader attempting t...
36  Art-house horror tries to use unconventional a...
37  I was completely bored with this film, melodra...
38  A rather lame teen slasher from Brisbane. Whil...
39  This was a decent movie for the first half. To...
40  Acolytes presents an interesting mix of origin...
41  I caught this on IFC last week and I thought i...
42  It's Saturday, it's raining, and I think every...
43  Frightmare begins with a horror movie icon kil...
44  Alas, another Costner movie that was an hour t...
45  My parents used to rent a lot of horror movies...
46  I wish I knew what to make of a movie like thi...
47  A tough sell: British playwright Ronald Harwoo...
48  Blake Edwards' legendary fiasco, begins to see...
49  I'm not a big fan of musicals, although this t...
50  David Bryce's comments nearby are exceptionall...

> Marcando a coluna de reviews negativos

In [20]:
df_test_neg.columns = ['reviews']
df_test_neg

reviews
0   Once again Mr. Costner has dragged out a movie...
1   Years ago, when DARLING LILI played on TV, it ...
2   Julie Andrews satirically prods her own goody-...
3   Blake Edwards tried very hard to change Julie ...
4   An obvious vanity press for Julie in her first...
5   Some may go for a film like this but I most as...
6   Few movies have dashed expectations and upset ...
7   For years, I've been a big fan of Park's work ...
8   Might end up being the biggest disappointment ...
9   Bloody Birthday opens to a shot of Meadowvale ...
10  This movie cannot be serious because it has a ...
11  This is a pale imitation of 'Officer and a Gen...
12  A mean spirited, repulsive horror film about 3...
13  Bloody Birthday is a totally rubbish slasher m...
14  When you make a film with a killer-kids premis...
15  When people say children are annoying u think ...
16  'Bloody Birthday' is an odd and, at times, hum...
17  Is it a poorly acted, cliche-ridden pile of tr...
18  Formulaic slasher film, only this one stars th...
19  Cover Girl is a lacklustre WWII musical with a...
20  Rita Hayworth plays a Brooklyn nightclub dance...
21  Like 'Singin' in the Rain', 'Cover Girl' has a...
22  It seems ever since 1982, about every two or t...
23  Rita Hayworth is just stunning at times and, f...
24  It hurt to watch this movie, it really did... ...
25  This film seems well made, and more efforts sh...
26  Sorry, but Jacqueline Hyde (get it??? - Jack L...
27  Jacqueline Hyde starts like any other normal d...
28  Maniratnam, who in India, is often compared wi...
29  This film, which I rented under the title "Bla...
30  The first 20 minutes were a little fun because...
31  I only came here to check Terror Hospital for ...
32  This is a hard film to rate. While it truly de...
33  Wow, another Kevin Costner hero movie. Postman...
34  I bought this on VHS as "Terror Hospital", and...
35  The film opens with a cult leader attempting t...
36  Art-house horror tries to use unconventional a...
37  I was completely bored with this film, melodra...
38  A rather lame teen slasher from Brisbane. Whil...
39  This was a decent movie for the first half. To...
40  Acolytes presents an interesting mix of origin...
41  I caught this on IFC last week and I thought i...
42  It's Saturday, it's raining, and I think every...
43  Frightmare begins with a horror movie icon kil...
44  Alas, another Costner movie that was an hour t...
45  My parents used to rent a lot of horror movies...
46  I wish I knew what to make of a movie like thi...
47  A tough sell: British playwright Ronald Harwoo...
48  Blake Edwards' legendary fiasco, begins to see...
49  I'm not a big fan of musicals, although this t...
50  David Bryce's comments nearby are exceptionall...

> marcando os dados de teste com o sentimento negativo

In [21]:
df_test_neg.loc[:,'sentiment'] = 'neg'
df_test_neg

reviews sentiment
0   Once again Mr. Costner has dragged out a movie...       neg
1   Years ago, when DARLING LILI played on TV, it ...       neg
2   Julie Andrews satirically prods her own goody-...       neg
3   Blake Edwards tried very hard to change Julie ...       neg
4   An obvious vanity press for Julie in her first...       neg
5   Some may go for a film like this but I most as...       neg
6   Few movies have dashed expectations and upset ...       neg
7   For years, I've been a big fan of Park's work ...       neg
8   Might end up being the biggest disappointment ...       neg
9   Bloody Birthday opens to a shot of Meadowvale ...       neg
10  This movie cannot be serious because it has a ...       neg
11  This is a pale imitation of 'Officer and a Gen...       neg
12  A mean spirited, repulsive horror film about 3...       neg
13  Bloody Birthday is a totally rubbish slasher m...       neg
14  When you make a film with a killer-kids premis...       neg
15  When people say children are annoying u think ...       neg
16  'Bloody Birthday' is an odd and, at times, hum...       neg
17  Is it a poorly acted, cliche-ridden pile of tr...       neg
18  Formulaic slasher film, only this one stars th...       neg
19  Cover Girl is a lacklustre WWII musical with a...       neg
20  Rita Hayworth plays a Brooklyn nightclub dance...       neg
21  Like 'Singin' in the Rain', 'Cover Girl' has a...       neg
22  It seems ever since 1982, about every two or t...       neg
23  Rita Hayworth is just stunning at times and, f...       neg
24  It hurt to watch this movie, it really did... ...       neg
25  This film seems well made, and more efforts sh...       neg
26  Sorry, but Jacqueline Hyde (get it??? - Jack L...       neg
27  Jacqueline Hyde starts like any other normal d...       neg
28  Maniratnam, who in India, is often compared wi...       neg
29  This film, which I rented under the title "Bla...       neg
30  The first 20 minutes were a little fun because...       neg
31  I only came here to check Terror Hospital for ...       neg
32  This is a hard film to rate. While it truly de...       neg
33  Wow, another Kevin Costner hero movie. Postman...       neg
34  I bought this on VHS as "Terror Hospital", and...       neg
35  The film opens with a cult leader attempting t...       neg
36  Art-house horror tries to use unconventional a...       neg
37  I was completely bored with this film, melodra...       neg
38  A rather lame teen slasher from Brisbane. Whil...       neg
39  This was a decent movie for the first half. To...       neg
40  Acolytes presents an interesting mix of origin...       neg
41  I caught this on IFC last week and I thought i...       neg
42  It's Saturday, it's raining, and I think every...       neg
43  Frightmare begins with a horror movie icon kil...       neg
44  Alas, another Costner movie that was an hour t...       neg
45  My parents used to rent a lot of horror movies...       neg
46  I wish I knew what to make of a movie like thi...       neg
47  A tough sell: British playwright Ronald Harwoo...       neg
48  Blake Edwards' legendary fiasco, begins to see...       neg
49  I'm not a big fan of musicals, although this t...       neg
50  David Bryce's comments nearby are exceptionall...       neg

### Teste - Sentimento POSITIVO

In [18]:
df_test_pos = pd.read_csv('https://drive.google.com/uc?id=1sk4SKAds3zUybi1J-TkVgiRf_NMZxmpF', encoding='utf8', header=None)
df_test_pos = df_test_pos.T # transpondo o dataframe para ficar no formato usual
df_test_pos

0
0   I went and saw this movie last night after bei...
1   If you had asked me how the movie was througho...
2   I was looking forward to The Guardian, but whe...
3   I was pleasantly surprised to find this movie ...
4   I work at a movie theater and every Thursday n...
5   I attended an advance screening of this film n...
6   As others that have commented around the web.....
7   I went to an advance screening of this movie t...
8   Brilliant and moving performances by Tom Court...
9   The Dresser is perhaps the most refined of bac...
10  This is one of my three all-time favorite movi...
11  My boyfriend and I went to watch The Guardian....
12  The many other comments about the film say it ...
13  The Dresser is a small but absolutely wonderfu...
14  What is worth mentioning that is omitted in th...
15  Anyone who appreciates fine acting and ringing...
16  Sir has played Lear over 200 times,but tonight...
17  This is a movie that deserves another look--if...
18  This film has its detractors, and Courtney's f...
19  Albert Finney and Tom Courtenay are brilliant ...
20  A fantastic cinema experience. I really enjoye...
21  <br /><br />What more can you ask for? A great...
22  My yardstick for measuring a movie's watch-abi...
23  'The Dresser' is one of those films which are ...
24  I just watched The Dresser this evening, havin...
25  LACKAWANNA BLUES is a fine stage play by Ruben...
26  Lackawanna Blues is a drama through and throug...
27  Lackawanna Blues is a touching story about Nan...
28  Every once in a while a film comes along with ...
29  LACKAWANNA BLUES is an entertaining, engrossin...
30  This is by far one of the best biographical fi...
31  S. Epatha Merkerson shines as Nanny in this to...
32  Ever since I first encountered the Divine Ms M...
33  How many movies are there that you can think o...
34  Lackawana Blues An impressive HBO movie about ...
35  What a wonderful film, filled with eccentric, ...
36  This is one of those movies that you just don'...
37  One of the best,Lackawanna Blues<br /><br />Gr...
38  Lackawanna Blues is a moving story about a boy...
39  Just a great soundtrack. Really enjoyable musi...
40  I have watch this movie almost every night tha...
41  Lackawanna Blues is and excellent movie. The c...
42  I watch Lackawanna Blues every time it comes o...
43  Lackawanna Blues is a very moving film depicti...
44  This movie was sadly under-promoted but proved...
45  Fantastically written, acted, and produced! Lo...
46  I only went to see this movie because I have a...
47  I was fortunate enough to see this movie on pr...
48  Our family (and the entire sold out sneak prev...
49  I'm a Petty Officer 1st Class (E-6) and have b...
50  I've seen this story before but my kids haven'...

> Marcando os dados da coluna de reviews

In [22]:
df_test_pos.columns = ['reviews']
df_test_pos

reviews
0   I went and saw this movie last night after bei...
1   If you had asked me how the movie was througho...
2   I was looking forward to The Guardian, but whe...
3   I was pleasantly surprised to find this movie ...
4   I work at a movie theater and every Thursday n...
5   I attended an advance screening of this film n...
6   As others that have commented around the web.....
7   I went to an advance screening of this movie t...
8   Brilliant and moving performances by Tom Court...
9   The Dresser is perhaps the most refined of bac...
10  This is one of my three all-time favorite movi...
11  My boyfriend and I went to watch The Guardian....
12  The many other comments about the film say it ...
13  The Dresser is a small but absolutely wonderfu...
14  What is worth mentioning that is omitted in th...
15  Anyone who appreciates fine acting and ringing...
16  Sir has played Lear over 200 times,but tonight...
17  This is a movie that deserves another look--if...
18  This film has its detractors, and Courtney's f...
19  Albert Finney and Tom Courtenay are brilliant ...
20  A fantastic cinema experience. I really enjoye...
21  <br /><br />What more can you ask for? A great...
22  My yardstick for measuring a movie's watch-abi...
23  'The Dresser' is one of those films which are ...
24  I just watched The Dresser this evening, havin...
25  LACKAWANNA BLUES is a fine stage play by Ruben...
26  Lackawanna Blues is a drama through and throug...
27  Lackawanna Blues is a touching story about Nan...
28  Every once in a while a film comes along with ...
29  LACKAWANNA BLUES is an entertaining, engrossin...
30  This is by far one of the best biographical fi...
31  S. Epatha Merkerson shines as Nanny in this to...
32  Ever since I first encountered the Divine Ms M...
33  How many movies are there that you can think o...
34  Lackawana Blues An impressive HBO movie about ...
35  What a wonderful film, filled with eccentric, ...
36  This is one of those movies that you just don'...
37  One of the best,Lackawanna Blues<br /><br />Gr...
38  Lackawanna Blues is a moving story about a boy...
39  Just a great soundtrack. Really enjoyable musi...
40  I have watch this movie almost every night tha...
41  Lackawanna Blues is and excellent movie. The c...
42  I watch Lackawanna Blues every time it comes o...
43  Lackawanna Blues is a very moving film depicti...
44  This movie was sadly under-promoted but proved...
45  Fantastically written, acted, and produced! Lo...
46  I only went to see this movie because I have a...
47  I was fortunate enough to see this movie on pr...
48  Our family (and the entire sold out sneak prev...
49  I'm a Petty Officer 1st Class (E-6) and have b...
50  I've seen this story before but my kids haven'...

> Marcando os reviews a seguir com o sentimento positivo.

In [23]:
df_test_pos.loc[:,'sentiment'] = 'pos'
df_test_pos

reviews sentiment
0   I went and saw this movie last night after bei...       pos
1   If you had asked me how the movie was througho...       pos
2   I was looking forward to The Guardian, but whe...       pos
3   I was pleasantly surprised to find this movie ...       pos
4   I work at a movie theater and every Thursday n...       pos
5   I attended an advance screening of this film n...       pos
6   As others that have commented around the web.....       pos
7   I went to an advance screening of this movie t...       pos
8   Brilliant and moving performances by Tom Court...       pos
9   The Dresser is perhaps the most refined of bac...       pos
10  This is one of my three all-time favorite movi...       pos
11  My boyfriend and I went to watch The Guardian....       pos
12  The many other comments about the film say it ...       pos
13  The Dresser is a small but absolutely wonderfu...       pos
14  What is worth mentioning that is omitted in th...       pos
15  Anyone who appreciates fine acting and ringing...       pos
16  Sir has played Lear over 200 times,but tonight...       pos
17  This is a movie that deserves another look--if...       pos
18  This film has its detractors, and Courtney's f...       pos
19  Albert Finney and Tom Courtenay are brilliant ...       pos
20  A fantastic cinema experience. I really enjoye...       pos
21  <br /><br />What more can you ask for? A great...       pos
22  My yardstick for measuring a movie's watch-abi...       pos
23  'The Dresser' is one of those films which are ...       pos
24  I just watched The Dresser this evening, havin...       pos
25  LACKAWANNA BLUES is a fine stage play by Ruben...       pos
26  Lackawanna Blues is a drama through and throug...       pos
27  Lackawanna Blues is a touching story about Nan...       pos
28  Every once in a while a film comes along with ...       pos
29  LACKAWANNA BLUES is an entertaining, engrossin...       pos
30  This is by far one of the best biographical fi...       pos
31  S. Epatha Merkerson shines as Nanny in this to...       pos
32  Ever since I first encountered the Divine Ms M...       pos
33  How many movies are there that you can think o...       pos
34  Lackawana Blues An impressive HBO movie about ...       pos
35  What a wonderful film, filled with eccentric, ...       pos
36  This is one of those movies that you just don'...       pos
37  One of the best,Lackawanna Blues<br /><br />Gr...       pos
38  Lackawanna Blues is a moving story about a boy...       pos
39  Just a great soundtrack. Really enjoyable musi...       pos
40  I have watch this movie almost every night tha...       pos
41  Lackawanna Blues is and excellent movie. The c...       pos
42  I watch Lackawanna Blues every time it comes o...       pos
43  Lackawanna Blues is a very moving film depicti...       pos
44  This movie was sadly under-promoted but proved...       pos
45  Fantastically written, acted, and produced! Lo...       pos
46  I only went to see this movie because I have a...       pos
47  I was fortunate enough to see this movie on pr...       pos
48  Our family (and the entire sold out sneak prev...       pos
49  I'm a Petty Officer 1st Class (E-6) and have b...       pos
50  I've seen this story before but my kids haven'...       pos

## Concatenando datasets de teste

In [26]:
df_test = pd.concat([df_test_neg, df_test_pos], axis=0, ignore_index=True)
df_test

reviews sentiment
0    Once again Mr. Costner has dragged out a movie...       neg
1    Years ago, when DARLING LILI played on TV, it ...       neg
2    Julie Andrews satirically prods her own goody-...       neg
3    Blake Edwards tried very hard to change Julie ...       neg
4    An obvious vanity press for Julie in her first...       neg
..                                                 ...       ...
97   I only went to see this movie because I have a...       pos
98   I was fortunate enough to see this movie on pr...       pos
99   Our family (and the entire sold out sneak prev...       pos
100  I'm a Petty Officer 1st Class (E-6) and have b...       pos
101  I've seen this story before but my kids haven'...       pos

[102 rows x 2 columns]

## Upload - Datasets de Treinamento

### Treinamento - Sentimento NEGATIVO

In [27]:
df_train_neg = pd.read_csv('https://drive.google.com/uc?id=1_y0TFz5lRyTWRPOdukm8jL4gmNBKtf6q', encoding='utf8', header=None)
df_train_neg = df_train_neg.T # transpondo o dataframe para ficar no formato usual
df_train_neg

0
0   Story of a man who has unnatural feelings for ...
1   This film had a lot of promise, and the plot w...
2   If you look at Corey Large's information here ...
3   All I could think of while watching this movie...
4   Weak plot, predictable violence, only semi int...
5   There are times when finishing a film one wish...
6   This is it. This is the one. This is the worst...
7   There are lots of extremely good-looking peopl...
8   Nine minutes of psychedelic, pulsating, often ...
9   This is an art film that was either made in 19...
10  I am a big fan of Arnold Vosloo. Finally seein...
11  Robert DeNiro plays the most unbelievably inte...
12  This movie was terrible. at first i just read ...
13  What was with all the Turkish actors? No offen...
14  I saw this DVD in my friends house and thought...
15  I rented this one on DVD without any prior kno...
16  I have to congratulate the genius who approved...
17  Sometimes a movie is so comprehensively awful ...
18  Assuming this won't end up a straight-to-video...
19  Terrible movie. Nuff Said.<br /><br />These Li...
20  This film is terrible. You don't really need t...
21  I cannot stay indifferent to Lars van Trier's ...
22  I saw the capsule comment said "great acting."...
23  I did not like the idea of the female turtle a...
24  This is the biggest insult to TMNT ever. Fortu...
25  Sometime in 1998, Saban had acquired the right...
26  This is the worst thing the TMNT franchise has...
27  Not only is it a disgustingly made low-budget ...
28  Sexo Cannibal, or Devil Hunter as it's more co...
29  This film seemed way too long even at only 75 ...
30  Devil Hunter gained notoriety for the fact tha...
31  or anyone who was praying for the sight of Al ...
32  An actress making a movie in Africa is kidnapp...
33  If I had not read Pat Barker's 'Union Street' ...
34  A model named Laura is working in South Americ...
35  How can you tell that a horror movie is terrib...
36  Of the three titles from Jess Franco to find t...
37  This video nasty was initially banned in Brita...
38  Not sure if I'm referring to those who labeled...
39  As the number of Video Nasties I've yet to see...
40  I have been looking for this film for ages bec...
41  When a man who doesn't have Alzheimer's can't ...
42  There are some nice shots in this film, it cat...
43  A very cheesy and dull road movie, with the in...
44  This fanciful horror flick has Vincent Price p...
45  Three part "horror" film with some guy in a bo...
46  I basically skimmed through the movie but just...
47  From the beginning of the movie, it gives the ...
48  This is really a new low in entertainment. Eve...
49  6/10 Acting, not great but some good acting.<b...
50  This is one of the dumbest films, I've ever se...

In [28]:
df_train_neg.columns = ['reviews']
df_train_neg

reviews
0   Story of a man who has unnatural feelings for ...
1   This film had a lot of promise, and the plot w...
2   If you look at Corey Large's information here ...
3   All I could think of while watching this movie...
4   Weak plot, predictable violence, only semi int...
5   There are times when finishing a film one wish...
6   This is it. This is the one. This is the worst...
7   There are lots of extremely good-looking peopl...
8   Nine minutes of psychedelic, pulsating, often ...
9   This is an art film that was either made in 19...
10  I am a big fan of Arnold Vosloo. Finally seein...
11  Robert DeNiro plays the most unbelievably inte...
12  This movie was terrible. at first i just read ...
13  What was with all the Turkish actors? No offen...
14  I saw this DVD in my friends house and thought...
15  I rented this one on DVD without any prior kno...
16  I have to congratulate the genius who approved...
17  Sometimes a movie is so comprehensively awful ...
18  Assuming this won't end up a straight-to-video...
19  Terrible movie. Nuff Said.<br /><br />These Li...
20  This film is terrible. You don't really need t...
21  I cannot stay indifferent to Lars van Trier's ...
22  I saw the capsule comment said "great acting."...
23  I did not like the idea of the female turtle a...
24  This is the biggest insult to TMNT ever. Fortu...
25  Sometime in 1998, Saban had acquired the right...
26  This is the worst thing the TMNT franchise has...
27  Not only is it a disgustingly made low-budget ...
28  Sexo Cannibal, or Devil Hunter as it's more co...
29  This film seemed way too long even at only 75 ...
30  Devil Hunter gained notoriety for the fact tha...
31  or anyone who was praying for the sight of Al ...
32  An actress making a movie in Africa is kidnapp...
33  If I had not read Pat Barker's 'Union Street' ...
34  A model named Laura is working in South Americ...
35  How can you tell that a horror movie is terrib...
36  Of the three titles from Jess Franco to find t...
37  This video nasty was initially banned in Brita...
38  Not sure if I'm referring to those who labeled...
39  As the number of Video Nasties I've yet to see...
40  I have been looking for this film for ages bec...
41  When a man who doesn't have Alzheimer's can't ...
42  There are some nice shots in this film, it cat...
43  A very cheesy and dull road movie, with the in...
44  This fanciful horror flick has Vincent Price p...
45  Three part "horror" film with some guy in a bo...
46  I basically skimmed through the movie but just...
47  From the beginning of the movie, it gives the ...
48  This is really a new low in entertainment. Eve...
49  6/10 Acting, not great but some good acting.<b...
50  This is one of the dumbest films, I've ever se...

In [29]:
df_train_neg.loc[:,'sentiment'] = 'neg'
df_train_neg

reviews sentiment
0   Story of a man who has unnatural feelings for ...       neg
1   This film had a lot of promise, and the plot w...       neg
2   If you look at Corey Large's information here ...       neg
3   All I could think of while watching this movie...       neg
4   Weak plot, predictable violence, only semi int...       neg
5   There are times when finishing a film one wish...       neg
6   This is it. This is the one. This is the worst...       neg
7   There are lots of extremely good-looking peopl...       neg
8   Nine minutes of psychedelic, pulsating, often ...       neg
9   This is an art film that was either made in 19...       neg
10  I am a big fan of Arnold Vosloo. Finally seein...       neg
11  Robert DeNiro plays the most unbelievably inte...       neg
12  This movie was terrible. at first i just read ...       neg
13  What was with all the Turkish actors? No offen...       neg
14  I saw this DVD in my friends house and thought...       neg
15  I rented this one on DVD without any prior kno...       neg
16  I have to congratulate the genius who approved...       neg
17  Sometimes a movie is so comprehensively awful ...       neg
18  Assuming this won't end up a straight-to-video...       neg
19  Terrible movie. Nuff Said.<br /><br />These Li...       neg
20  This film is terrible. You don't really need t...       neg
21  I cannot stay indifferent to Lars van Trier's ...       neg
22  I saw the capsule comment said "great acting."...       neg
23  I did not like the idea of the female turtle a...       neg
24  This is the biggest insult to TMNT ever. Fortu...       neg
25  Sometime in 1998, Saban had acquired the right...       neg
26  This is the worst thing the TMNT franchise has...       neg
27  Not only is it a disgustingly made low-budget ...       neg
28  Sexo Cannibal, or Devil Hunter as it's more co...       neg
29  This film seemed way too long even at only 75 ...       neg
30  Devil Hunter gained notoriety for the fact tha...       neg
31  or anyone who was praying for the sight of Al ...       neg
32  An actress making a movie in Africa is kidnapp...       neg
33  If I had not read Pat Barker's 'Union Street' ...       neg
34  A model named Laura is working in South Americ...       neg
35  How can you tell that a horror movie is terrib...       neg
36  Of the three titles from Jess Franco to find t...       neg
37  This video nasty was initially banned in Brita...       neg
38  Not sure if I'm referring to those who labeled...       neg
39  As the number of Video Nasties I've yet to see...       neg
40  I have been looking for this film for ages bec...       neg
41  When a man who doesn't have Alzheimer's can't ...       neg
42  There are some nice shots in this film, it cat...       neg
43  A very cheesy and dull road movie, with the in...       neg
44  This fanciful horror flick has Vincent Price p...       neg
45  Three part "horror" film with some guy in a bo...       neg
46  I basically skimmed through the movie but just...       neg
47  From the beginning of the movie, it gives the ...       neg
48  This is really a new low in entertainment. Eve...       neg
49  6/10 Acting, not great but some good acting.<b...       neg
50  This is one of the dumbest films, I've ever se...       neg

### Treinamento - Sentimento POSITIVO

In [30]:
df_train_pos = pd.read_csv('https://drive.google.com/uc?id=1HCybdmPv4gOAI2kQCb5lz8aTMDTNVrr0', encoding='utf8', header=None)
df_train_pos = df_train_pos.T # transpondo o dataframe para ficar no formato usual
df_train_pos

0
0   Bromwell High is a cartoon comedy. It ran at t...
1   I'm a male, not given to women's movies, but t...
2   Liked Stanley & Iris very much. Acting was ver...
3   Liked Stanley & Iris very much. Acting was ver...
4   The production quality, cast, premise, authent...
5   This film has a special place in my heart, as ...
6   I guess if a film has magic, I don't need it t...
7   I found this to be a so-so romance/drama that ...
8   This is a complex film that explores the effec...
9   `Stanley and Iris' is a heart warming film abo...
10  I just read the comments of TomReynolds2004 an...
11  If you like adult comedy cartoons, like South ...
12  'Stanley and Iris' show the triumph of the hum...
13  This may not be a memorable classic, but it is...
14  If you fast forward through the horrible singi...
15  OK, so the musical pieces were poorly written ...
16  Although this was obviously a low-budget produ...
17  I never saw this when I was a kid, so this was...
18  When I was a kid I watched this many times ove...
19  Emilio Miraglio's "The Red Queen Kills Seven T...
20  And here's yet another piece of evidence to cl...
21  There are some films that every Horror fan owe...
22  Bromwell High is nothing short of brilliant. E...
23  An old family story told to two young girls by...
24  Once upon a time in a castle...... Two little ...
25  Emilio Miraglia's first Giallo feature, The Ni...
26  This interesting Giallo boosts a typical but s...
27  What a fun movie! If you're a Giallo fan, Red ...
28  When I saw the elaborate DVD box for this and ...
29  Another good Stooge short!Christine McIntyre i...
30  Of the elements that make this the best at thi...
31  One of the very best Three Stooges shorts ever...
32  This is apparently one of Shemp's first shorts...
33  All the world's a stage and its people actors ...
34  An ultra-nervous old man, "Mr. Goodrich," terr...
35  Who Done It? contains many surefire laughs and...
36  Busy Phillips put in one hell of a performance...
37  I liked this movie a lot. It really intrigued ...
38  When i was told of this movie i thought it wou...
39  Home Room was a great movie if you've ever had...
40  Just finished watching this movie for maybe th...
41  It breaks my heart that this movie is not appr...
42  I'm from Belgium and therefore my English writ...
43  If you "get it", it's magnificent.<br /><br />...
44  FUTZ is the only show preserved from the exper...
45  My children just happened to stop at this movi...
46  I came in in the middle of this film so I had ...
47  Fair drama/love story movie that focuses on th...
48  Although I didn't like Stanley & Iris tremendo...
49  Very good drama although it appeared to have a...
50  Working-class romantic drama from director Mar...

In [31]:
df_train_pos.columns = ['reviews']
df_train_pos

reviews
0   Bromwell High is a cartoon comedy. It ran at t...
1   I'm a male, not given to women's movies, but t...
2   Liked Stanley & Iris very much. Acting was ver...
3   Liked Stanley & Iris very much. Acting was ver...
4   The production quality, cast, premise, authent...
5   This film has a special place in my heart, as ...
6   I guess if a film has magic, I don't need it t...
7   I found this to be a so-so romance/drama that ...
8   This is a complex film that explores the effec...
9   `Stanley and Iris' is a heart warming film abo...
10  I just read the comments of TomReynolds2004 an...
11  If you like adult comedy cartoons, like South ...
12  'Stanley and Iris' show the triumph of the hum...
13  This may not be a memorable classic, but it is...
14  If you fast forward through the horrible singi...
15  OK, so the musical pieces were poorly written ...
16  Although this was obviously a low-budget produ...
17  I never saw this when I was a kid, so this was...
18  When I was a kid I watched this many times ove...
19  Emilio Miraglio's "The Red Queen Kills Seven T...
20  And here's yet another piece of evidence to cl...
21  There are some films that every Horror fan owe...
22  Bromwell High is nothing short of brilliant. E...
23  An old family story told to two young girls by...
24  Once upon a time in a castle...... Two little ...
25  Emilio Miraglia's first Giallo feature, The Ni...
26  This interesting Giallo boosts a typical but s...
27  What a fun movie! If you're a Giallo fan, Red ...
28  When I saw the elaborate DVD box for this and ...
29  Another good Stooge short!Christine McIntyre i...
30  Of the elements that make this the best at thi...
31  One of the very best Three Stooges shorts ever...
32  This is apparently one of Shemp's first shorts...
33  All the world's a stage and its people actors ...
34  An ultra-nervous old man, "Mr. Goodrich," terr...
35  Who Done It? contains many surefire laughs and...
36  Busy Phillips put in one hell of a performance...
37  I liked this movie a lot. It really intrigued ...
38  When i was told of this movie i thought it wou...
39  Home Room was a great movie if you've ever had...
40  Just finished watching this movie for maybe th...
41  It breaks my heart that this movie is not appr...
42  I'm from Belgium and therefore my English writ...
43  If you "get it", it's magnificent.<br /><br />...
44  FUTZ is the only show preserved from the exper...
45  My children just happened to stop at this movi...
46  I came in in the middle of this film so I had ...
47  Fair drama/love story movie that focuses on th...
48  Although I didn't like Stanley & Iris tremendo...
49  Very good drama although it appeared to have a...
50  Working-class romantic drama from director Mar...

In [32]:
df_train_pos.loc[:,'sentiment'] = 'pos'
df_train_pos

reviews sentiment
0   Bromwell High is a cartoon comedy. It ran at t...       pos
1   I'm a male, not given to women's movies, but t...       pos
2   Liked Stanley & Iris very much. Acting was ver...       pos
3   Liked Stanley & Iris very much. Acting was ver...       pos
4   The production quality, cast, premise, authent...       pos
5   This film has a special place in my heart, as ...       pos
6   I guess if a film has magic, I don't need it t...       pos
7   I found this to be a so-so romance/drama that ...       pos
8   This is a complex film that explores the effec...       pos
9   `Stanley and Iris' is a heart warming film abo...       pos
10  I just read the comments of TomReynolds2004 an...       pos
11  If you like adult comedy cartoons, like South ...       pos
12  'Stanley and Iris' show the triumph of the hum...       pos
13  This may not be a memorable classic, but it is...       pos
14  If you fast forward through the horrible singi...       pos
15  OK, so the musical pieces were poorly written ...       pos
16  Although this was obviously a low-budget produ...       pos
17  I never saw this when I was a kid, so this was...       pos
18  When I was a kid I watched this many times ove...       pos
19  Emilio Miraglio's "The Red Queen Kills Seven T...       pos
20  And here's yet another piece of evidence to cl...       pos
21  There are some films that every Horror fan owe...       pos
22  Bromwell High is nothing short of brilliant. E...       pos
23  An old family story told to two young girls by...       pos
24  Once upon a time in a castle...... Two little ...       pos
25  Emilio Miraglia's first Giallo feature, The Ni...       pos
26  This interesting Giallo boosts a typical but s...       pos
27  What a fun movie! If you're a Giallo fan, Red ...       pos
28  When I saw the elaborate DVD box for this and ...       pos
29  Another good Stooge short!Christine McIntyre i...       pos
30  Of the elements that make this the best at thi...       pos
31  One of the very best Three Stooges shorts ever...       pos
32  This is apparently one of Shemp's first shorts...       pos
33  All the world's a stage and its people actors ...       pos
34  An ultra-nervous old man, "Mr. Goodrich," terr...       pos
35  Who Done It? contains many surefire laughs and...       pos
36  Busy Phillips put in one hell of a performance...       pos
37  I liked this movie a lot. It really intrigued ...       pos
38  When i was told of this movie i thought it wou...       pos
39  Home Room was a great movie if you've ever had...       pos
40  Just finished watching this movie for maybe th...       pos
41  It breaks my heart that this movie is not appr...       pos
42  I'm from Belgium and therefore my English writ...       pos
43  If you "get it", it's magnificent.<br /><br />...       pos
44  FUTZ is the only show preserved from the exper...       pos
45  My children just happened to stop at this movi...       pos
46  I came in in the middle of this film so I had ...       pos
47  Fair drama/love story movie that focuses on th...       pos
48  Although I didn't like Stanley & Iris tremendo...       pos
49  Very good drama although it appeared to have a...       pos
50  Working-class romantic drama from director Mar...       pos

### Treinamento - UNSUPOSED

In [34]:
df_train_unsup = pd.read_csv('https://drive.google.com/uc?id=1KDFnhw4Syj9I9PFfcNSQTZWu8Ohi907C', encoding='utf8', header=None)
df_train_unsup = df_train_unsup.T # transpondo o dataframe para ficar no formato usual
df_train_unsup

0
0   I admit, the great majority of films released ...
1   Finally, I got to see this legendary masterpie...
2   Watching this film was like reliving the jobs ...
3   Evocative, funny look at four women stuck in t...
4   I thought I'd do a quick review. Actually I di...
5   Clockwatchers surprised me in many ways. At fi...
6   Transcendently beautiful in moments outside th...
7   I have been an office temp, and this portrays ...
8   I've been reading the comments and I'm not sur...
9   Clockwatchers is a very funny little film that...
10  This was clearly written by someone who has go...
11  This is a very strange film that was long thou...
12  For some reason I'd been resisting seeing this...
13  Funny women Posey and Kudrow star in this popu...
14  I have never worked as an office temp before, ...
15  The four leads are very effective as the disaf...
16  Even if it is the last movie in the video stor...
17  This film is about the minutiae of office work...
18  Could call this four actresses in search of a ...
19  This is a very bleak film...if you're looking ...
20  That is exactly what I did while watching this...
21  Many of you have posted comments on "Clockwatc...
22  William Boyd and Louis Wolheim are the "Two Ar...
23  I enjoyed this movie. I rented it because it h...
24  With the movie market being so saturated with ...
25  I'm surprised how many people see this as a bo...
26  When the credits started for "Clockwatchers" -...
27  Is this movie slow? Occasionally boring? Kind ...
28  Blistering black comedy co-written by Jill Spr...
29  I saw this on cable last night, just 2 days af...
30  Jill Sprecher, the enormous talent behind "Clo...
31  Some of the earlier reviewers have called "Clo...
32  I suppose the viewers who label Clockwatchers ...
33  Not very many movies come to my mind that cove...
34  The only thing that I can think when reading t...
35  Iraq was in a terrible way LONG before the Ame...
36  I got this movie in the mail a few days ago. I...
37  This was a waste of time. Out of context video...
38  I found it hard to view the user comments on t...
39  I found "Lonely Man' in a local library. I can...
40  This film has got merit not least the photogra...
41  This isn't the cowboys vs. indians or "saved b...
42  This film has a lot going for it. It was done ...
43  Since childhood I've found the actors playing ...
44  The third movie produced by Howard Hughes, thi...
45  Jacob Wade (Jack Palance) used to be a celebra...
46  Turner Classic Movies showed this silent, B&W ...
47  Enjoyed viewing this 1927 Silent Film from the...
48  Like TWO ARABIAN KNIGHTS, a couple of American...
49  TWO ARABIAN KNIGHTS has been restored by Flick...
50  TCM came through by acquiring this wonderful, ...

In [35]:
df_train_unsup.columns = ['reviews']
df_train_unsup

reviews
0   I admit, the great majority of films released ...
1   Finally, I got to see this legendary masterpie...
2   Watching this film was like reliving the jobs ...
3   Evocative, funny look at four women stuck in t...
4   I thought I'd do a quick review. Actually I di...
5   Clockwatchers surprised me in many ways. At fi...
6   Transcendently beautiful in moments outside th...
7   I have been an office temp, and this portrays ...
8   I've been reading the comments and I'm not sur...
9   Clockwatchers is a very funny little film that...
10  This was clearly written by someone who has go...
11  This is a very strange film that was long thou...
12  For some reason I'd been resisting seeing this...
13  Funny women Posey and Kudrow star in this popu...
14  I have never worked as an office temp before, ...
15  The four leads are very effective as the disaf...
16  Even if it is the last movie in the video stor...
17  This film is about the minutiae of office work...
18  Could call this four actresses in search of a ...
19  This is a very bleak film...if you're looking ...
20  That is exactly what I did while watching this...
21  Many of you have posted comments on "Clockwatc...
22  William Boyd and Louis Wolheim are the "Two Ar...
23  I enjoyed this movie. I rented it because it h...
24  With the movie market being so saturated with ...
25  I'm surprised how many people see this as a bo...
26  When the credits started for "Clockwatchers" -...
27  Is this movie slow? Occasionally boring? Kind ...
28  Blistering black comedy co-written by Jill Spr...
29  I saw this on cable last night, just 2 days af...
30  Jill Sprecher, the enormous talent behind "Clo...
31  Some of the earlier reviewers have called "Clo...
32  I suppose the viewers who label Clockwatchers ...
33  Not very many movies come to my mind that cove...
34  The only thing that I can think when reading t...
35  Iraq was in a terrible way LONG before the Ame...
36  I got this movie in the mail a few days ago. I...
37  This was a waste of time. Out of context video...
38  I found it hard to view the user comments on t...
39  I found "Lonely Man' in a local library. I can...
40  This film has got merit not least the photogra...
41  This isn't the cowboys vs. indians or "saved b...
42  This film has a lot going for it. It was done ...
43  Since childhood I've found the actors playing ...
44  The third movie produced by Howard Hughes, thi...
45  Jacob Wade (Jack Palance) used to be a celebra...
46  Turner Classic Movies showed this silent, B&W ...
47  Enjoyed viewing this 1927 Silent Film from the...
48  Like TWO ARABIAN KNIGHTS, a couple of American...
49  TWO ARABIAN KNIGHTS has been restored by Flick...
50  TCM came through by acquiring this wonderful, ...

In [36]:
df_train_unsup.loc[:,'sentiment'] = 'unsup'
df_train_unsup

reviews sentiment
0   I admit, the great majority of films released ...     unsup
1   Finally, I got to see this legendary masterpie...     unsup
2   Watching this film was like reliving the jobs ...     unsup
3   Evocative, funny look at four women stuck in t...     unsup
4   I thought I'd do a quick review. Actually I di...     unsup
5   Clockwatchers surprised me in many ways. At fi...     unsup
6   Transcendently beautiful in moments outside th...     unsup
7   I have been an office temp, and this portrays ...     unsup
8   I've been reading the comments and I'm not sur...     unsup
9   Clockwatchers is a very funny little film that...     unsup
10  This was clearly written by someone who has go...     unsup
11  This is a very strange film that was long thou...     unsup
12  For some reason I'd been resisting seeing this...     unsup
13  Funny women Posey and Kudrow star in this popu...     unsup
14  I have never worked as an office temp before, ...     unsup
15  The four leads are very effective as the disaf...     unsup
16  Even if it is the last movie in the video stor...     unsup
17  This film is about the minutiae of office work...     unsup
18  Could call this four actresses in search of a ...     unsup
19  This is a very bleak film...if you're looking ...     unsup
20  That is exactly what I did while watching this...     unsup
21  Many of you have posted comments on "Clockwatc...     unsup
22  William Boyd and Louis Wolheim are the "Two Ar...     unsup
23  I enjoyed this movie. I rented it because it h...     unsup
24  With the movie market being so saturated with ...     unsup
25  I'm surprised how many people see this as a bo...     unsup
26  When the credits started for "Clockwatchers" -...     unsup
27  Is this movie slow? Occasionally boring? Kind ...     unsup
28  Blistering black comedy co-written by Jill Spr...     unsup
29  I saw this on cable last night, just 2 days af...     unsup
30  Jill Sprecher, the enormous talent behind "Clo...     unsup
31  Some of the earlier reviewers have called "Clo...     unsup
32  I suppose the viewers who label Clockwatchers ...     unsup
33  Not very many movies come to my mind that cove...     unsup
34  The only thing that I can think when reading t...     unsup
35  Iraq was in a terrible way LONG before the Ame...     unsup
36  I got this movie in the mail a few days ago. I...     unsup
37  This was a waste of time. Out of context video...     unsup
38  I found it hard to view the user comments on t...     unsup
39  I found "Lonely Man' in a local library. I can...     unsup
40  This film has got merit not least the photogra...     unsup
41  This isn't the cowboys vs. indians or "saved b...     unsup
42  This film has a lot going for it. It was done ...     unsup
43  Since childhood I've found the actors playing ...     unsup
44  The third movie produced by Howard Hughes, thi...     unsup
45  Jacob Wade (Jack Palance) used to be a celebra...     unsup
46  Turner Classic Movies showed this silent, B&W ...     unsup
47  Enjoyed viewing this 1927 Silent Film from the...     unsup
48  Like TWO ARABIAN KNIGHTS, a couple of American...     unsup
49  TWO ARABIAN KNIGHTS has been restored by Flick...     unsup
50  TCM came through by acquiring this wonderful, ...     unsup

## Concatenando datasets de treino

In [37]:
df_train = pd.concat([df_train_neg, df_train_pos, df_train_unsup], axis=0, ignore_index=True)
df_train

reviews sentiment
0    Story of a man who has unnatural feelings for ...       neg
1    This film had a lot of promise, and the plot w...       neg
2    If you look at Corey Large's information here ...       neg
3    All I could think of while watching this movie...       neg
4    Weak plot, predictable violence, only semi int...       neg
..                                                 ...       ...
148  Turner Classic Movies showed this silent, B&W ...     unsup
149  Enjoyed viewing this 1927 Silent Film from the...     unsup
150  Like TWO ARABIAN KNIGHTS, a couple of American...     unsup
151  TWO ARABIAN KNIGHTS has been restored by Flick...     unsup
152  TCM came through by acquiring this wonderful, ...     unsup

[153 rows x 2 columns]

## Concatenando datasets totais

> O dataset abaixo tem o formato semelhante ao dataset do IMDB no link:
[IMDB reviews dataset](https://www.kaggle.com/lakshmi25npathi/imdb-dataset-of-50k-movie-reviews).

> Para que a extração demorasse menos tempo, optou-se por fazer um exemplo menor com as 50 entradas de cada pasta, conforme o branch [Data-Extraction](https://github.com/dgavieira/NLP-Celero/tree/data-extraction) no repositório.

In [38]:
df_total = pd.concat([df_train, df_test], axis=0, ignore_index=True)
df_total

reviews sentiment
0    Story of a man who has unnatural feelings for ...       neg
1    This film had a lot of promise, and the plot w...       neg
2    If you look at Corey Large's information here ...       neg
3    All I could think of while watching this movie...       neg
4    Weak plot, predictable violence, only semi int...       neg
..                                                 ...       ...
250  I only went to see this movie because I have a...       pos
251  I was fortunate enough to see this movie on pr...       pos
252  Our family (and the entire sold out sneak prev...       pos
253  I'm a Petty Officer 1st Class (E-6) and have b...       pos
254  I've seen this story before but my kids haven'...       pos

[255 rows x 2 columns]

# Verificando IMDB Dataset of 50k Movie Reviews

In [40]:
df_imdb = pd.read_csv('https://drive.google.com/uc?id=1dYzliUyfAU3jepZpUeBJE16elVxOYZHl', encoding='utf8')
df_imdb

review sentiment
0      One of the other reviewers has mentioned that ...  positive
1      A wonderful little production. <br /><br />The...  positive
2      I thought this was a wonderful way to spend ti...  positive
3      Basically there's a family where a little boy ...  negative
4      Petter Mattei's "Love in the Time of Money" is...  positive
...                                                  ...       ...
49995  I thought this movie did a down right good job...  positive
49996  Bad plot, bad dialogue, bad acting, idiotic di...  negative
49997  I am a Catholic taught in parochial elementary...  negative
49998  I'm going to have to disagree with the previou...  negative
49999  No one expects the Star Trek movies to be high...  negative

[50000 rows x 2 columns]

> Observa-se que os datasets são semelhantes, portanto, para fins de análise exploratória, posso utilizar a variável df_imdb para os itens a seguir.

# Análise Exploratória

In [42]:
from sklearn import linear_model
from sklearn import svm
from sklearn.model_selection import train_test_split
from sklearn.feature_extraction.text import CountVectorizer

## Mudando classificação de [negative, positive] para [0,1]

In [43]:
classificacao = df_imdb["sentiment"].replace(["negative", "positive"],[0,1])
df_imdb["classificacao"] = classificacao
df_imdb.head()

review sentiment  classificacao
0  One of the other reviewers has mentioned that ...  positive              1
1  A wonderful little production. <br /><br />The...  positive              1
2  I thought this was a wonderful way to spend ti...  positive              1
3  Basically there's a family where a little boy ...  negative              0
4  Petter Mattei's "Love in the Time of Money" is...  positive              1

In [44]:
df_imdb.tail()

review  ... classificacao
49995  I thought this movie did a down right good job...  ...             1
49996  Bad plot, bad dialogue, bad acting, idiotic di...  ...             0
49997  I am a Catholic taught in parochial elementary...  ...             0
49998  I'm going to have to disagree with the previou...  ...             0
49999  No one expects the Star Trek movies to be high...  ...             0

[5 rows x 3 columns]

In [45]:
print(df_imdb['classificacao'].value_counts())

1    25000
0    25000
Name: classificacao, dtype: int64


> observo que os dados são perfeitamente equilibrados.

* 25000 dados marcados como negativos.
* 25000 dados marcados como positivos.

## Vetorizando palavras em um bag of words

In [46]:
from sklearn.feature_extraction.text import CountVectorizer

In [47]:
vetorizar = CountVectorizer(lowercase=False, max_features=50)
bag_of_words = vetorizar.fit_transform(df_imdb['review'])

In [48]:
print(bag_of_words.shape)

(50000, 50)


In [49]:
vetorizar.get_feature_names()

['It',
 'The',
 'This',
 'about',
 'all',
 'an',
 'and',
 'are',
 'as',
 'at',
 'be',
 'br',
 'but',
 'by',
 'can',
 'film',
 'for',
 'from',
 'good',
 'has',
 'have',
 'he',
 'her',
 'his',
 'in',
 'is',
 'it',
 'just',
 'like',
 'more',
 'movie',
 'not',
 'of',
 'on',
 'one',
 'or',
 'out',
 'so',
 'some',
 'that',
 'the',
 'they',
 'this',
 'to',
 'up',
 'very',
 'was',
 'who',
 'with',
 'you']

In [50]:
vetorizado = pd.DataFrame.sparse.from_spmatrix(bag_of_words, columns = vetorizar.get_feature_names())
vetorizado

It  The  This  about  all  an  and  ...  to  up  very  was  who  with  you
0       2    2     1      1    1   1    6  ...   6   0     0    3    2     5    3
1       1    3     0      1    2   0    7  ...   2   0     2    0    0     3    0
2       0    1     2      0    0   0    4  ...   4   0     0    4    0     2    0
3       0    0     1      0    3   0    3  ...   3   0     0    0    0     3    2
4       0    2     2      2    2   0    5  ...   7   0     0    0    0     1    0
...    ..  ...   ...    ...  ...  ..  ...  ...  ..  ..   ...  ...  ...   ...  ...
49995   2    0     0      1    1   0    7  ...   5   0     1    2    2     0    2
49996   1    3     0      0    1   0    2  ...   3   0     0    4    1     0    1
49997   0    0     1      0    1   1    6  ...   4   0     1    1    1     1    0
49998   1    0     1      2    1   0    8  ...   6   0     0    2    0     3    1
49999   0    0     0      1    1   0    2  ...   3   0     0    0    1     1    1

[50000 rows x 50 columns]

## Classificando sentimentos com Linear Regression

In [52]:
from sklearn.model_selection import train_test_split

In [53]:
treino, teste, classe_treino, classe_teste = train_test_split(bag_of_words,
                                                              df_imdb['classificacao'],
                                                              random_state=42)

In [54]:
print(treino.shape)
print(teste.shape)

(37500, 50)
(12500, 50)


### Mostrando os resultados em termos de probabilidade

In [55]:
regressao_logistica = LogisticRegression(solver='lbfgs')
regressao_logistica.fit(treino, classe_treino)
previsao_teste = regressao_logistica.predict_proba(teste)
print(previsao_teste)

[[0.59665069 0.40334931]
 [0.45141023 0.54858977]
 [0.68065739 0.31934261]
 ...
 [0.54112719 0.45887281]
 [0.24831594 0.75168406]
 [0.5179249  0.4820751 ]]


### Mostrando os resultados em termos booleanos

In [56]:
previsao_test_bol = previsao_teste[:,1] >= 0.5
print(previsao_test_bol)

[False  True False ... False  True False]


### Mostrando os resultados em termos binários

In [58]:
import numpy as np

In [59]:
previsao_teste_int = previsao_test_bol.astype(np.int)
print(previsao_teste_int)

[0 1 0 ... 0 1 0]


### Mostrando os resultados em termos de porcentagem

In [61]:
from sklearn.metrics import accuracy_score
acuracia_teste = accuracy_score(classe_teste, previsao_teste_int)
print("Precisão do Modelo: {:.2f}%".format(acuracia_teste*100))

Precisão do Modelo: 65.20%


> Observo que o modelo treinado por Regressão Logística sem nenhum tratamento de NLP atinge 65.20% de acurácia. É possível melhorar a performance utilizando técnicas de processamento de linguagem natural - o recomendável - ou mudando o modelo. Para fazer o teste, irei verificar utilizando uma SVM Linear.

## Classificando com SVM

In [62]:
svm_linearSVC = svm.LinearSVC()
svm_linearSVC.fit(treino, classe_treino)

/usr/local/lib/python3.6/dist-packages/sklearn/svm/_base.py:947: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  "the number of iterations.", ConvergenceWarning)


LinearSVC(C=1.0, class_weight=None, dual=True, fit_intercept=True,
          intercept_scaling=1, loss='squared_hinge', max_iter=1000,
          multi_class='ovr', penalty='l2', random_state=None, tol=0.0001,
          verbose=0)

In [63]:
svm_linearSVC.predict(teste)

array([0, 1, 0, ..., 0, 1, 0])

In [67]:
score_svm = svm_linearSVC.score(teste, classe_teste)
print("Precisão do Modelo: {:.2f}%".format(score_svm*100))

Precisão do Modelo: 65.16%


## Comparando modelos de Regressão Logística e LinearSVC

In [68]:
print("Precisão do Modelo Linear Regression: {:.2f}%".format(acuracia_teste*100))
print("Precisão do Modelo LinearSVC: {:.2f}%".format(score_svm*100))

Precisão do Modelo Linear Regression: 65.20%
Precisão do Modelo LinearSVC: 65.16%


# Executando técnicas de Natural Language Processing